common/autoware_motion_utils/include/autoware/motion_utils/trajectory_container/interpolator/nearest_neighbor.hpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__MOTION_UTILS__TRAJECTORY_CONTAINER__INTERPOLATOR__NEAREST_NEIGHBOR_HPP_
+#define AUTOWARE__MOTION_UTILS__TRAJECTORY_CONTAINER__INTERPOLATOR__NEAREST_NEIGHBOR_HPP_
+
+#include "autoware/motion_utils/trajectory_container/interpolator/interpolator.hpp"
+
+#include <vector>
+
+namespace autoware::motion_utils::trajectory_container::interpolator
+{
+
+/**
+ * @brief Template class for nearest neighbor interpolation.
+ *
+ * This class provides methods to perform nearest neighbor interpolation on a set of data points.
+ *
+ * @tparam T The type of the values being interpolated.
+ */
+template <typename T>
+class NearestNeighbor;
+
+namespace detail
+{
+
+/**
+ * @brief Common Implementation of nearest neighbor.
+ *
+ * This class implements the core functionality for nearest neighbor interpolation.
+ *
+ * @tparam T The type of the values being interpolated.
+ */
+template <typename T>
+class NearestNeighborCommonImpl : public Interpolator<T>
+{
+protected:
+  std::vector<T> values_;  ///< Interpolation values.
+
+  /**
+   * @brief Compute the interpolated value at the given point.
+   *
+   * @param s The point at which to compute the interpolated value.
+   * @return The interpolated value.
+   */
+  [[nodiscard]] T compute_impl(const double & s) const override
+  {
+    int idx = this->get_index(s);
```

Comment:
@yhisaki 

Can you update all `int` ? 🙏

Reply:
Sorry, I missed it....
I will modify!

Reply:
Thanks a lot!! :+1: :cool:

Reply:
fixed!

Reply:
Thanks!
Is it okay to leave `static_cast<int>` as `int` ?

Reply:
Sorry for making the same mistake over and over again .... I overlooked it.

Reply:
No, problem!! Thanks :smile:

Reply:
I think I have fixed all the problems!